src/functional.rs

```diff
@@ -122,13 +106,9 @@ impl<K, V> Tree<K, V> {
     where
         K: cmp::Ord,
     {
-        match self {
-            Tree::Leaf => None,
-            Tree::Node(n) => match k.cmp(&n.key) {
-                cmp::Ordering::Less => n.left.find(k),
-                cmp::Ordering::Equal => Some(&n.value),
-                cmp::Ordering::Greater => n.right.find(k),
-            },
+        match &self.root {
+            None => None,
+            Some(n) => n.find(k),
         }
```

Comment:
This, and many other methods, could conceivably leverage something like
```rust
self.root.as_ref().and_then(|n| n.find(k))
```
but, in benchmarks, that appears to be slower. Not sure if `std` needs a `map_ref` haha